### PR TITLE
HO 43 OrgUser and IndUser can add comments onto a request/listing

### PR DESCRIPTION
**Pull request by tlchambers**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,7 +1,30 @@
...
+## To run the application 
+2. Start by running the backend server `npm run dev`
+  ```
+  ; cd backend
+  ; npm run dev
+  ```
+3. Then cd into the frontend folder and run `npm start`
+  ```
+  ; cd frontend
+  ; npm run start
```
Comment (Sabirah42) [inline README.md:20]: Line 17 and 20 have different instructions - should 'run' be included?

Comment (tlchambers) [reply README.md:20]: Yes you're absolutely right! Will action that

File: README.md [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+#### The Backend (API)
+**Note Environment variables (e.g. JWT secret) are set in the .env. file**
 
-To run the backend cd to the backend folder and then run: npm run dev
-
-For the frontend cd to the frontend folder and then run: npm start
+4. To start the server in test mode (so that it connects to the test DB)
+  ```
+  ; cd backend
+  ; npm run test
```
Comment (Sabirah42) [inline README.md:29]: When you demo'd you just did 'npm test' - does this make a diff?

Comment (tlchambers) [reply README.md:29]: Crap you're right, it's npm test! Amending it now

File: backend/controllers/listingsController.js [hunk 3]
```diff
@@ -1,41 +1,84 @@
...
+  const listing = await Listing.findByIdAndRemove(id);
+
+  if (!listing) {
+    return res.status(404).send(`No listing with id: ${id}`);
+  }
+
+  res.json({ message: "Listing deleted successfully." });
+};
+
+const addCommentToAListing = async (req, res) => {
```
Comment (Sabirah42) [inline backend/controllers/listingsController.js:50]: Placeholder comment whilst we read more Comments code

File: backend/controllers/orgUsersController.js [hunk 4]
```diff
@@ -1,20 +1,58 @@
 const OrgUser = require("../models/orgUserModel");
-const mongoose = require('mongoose');
+const jwt = require("jsonwebtoken");
 
-const getAllOrgUsers = async (req, res) => {
```
Comment (Sabirah42) [inline backend/controllers/orgUsersController.js]: Can we leave this in for now for testing in Postman?

File: backend/controllers/orgUsersController.js [hunk 5]
```diff
@@ -1,20 +1,58 @@
...
+    let orgUser = req.params.orgUser_id;
+    res.status(200).json({ orgUser });
+  } catch (error) {
+    res.status(400).json({ error: error.message });
   }
-}
+};
 
 module.exports = {
-  getAllOrgUsers, createOrgUser
```
Comment (Sabirah42) [inline backend/controllers/orgUsersController.js]: Same again here re. 'get all' method

File: backend/controllers/orgUsersController.js (hunk 4 above)
Comment (tlchambers) [reply backend/controllers/orgUsersController.js]: Yes sorr

File: backend/models/commentModel.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const mongoose = require("mongoose");
+mongoose.set("strictQuery", true);
+const validator = require("validator");
+
+const OrgUser = require("../models/orgUserModel");
+
+const CommentSchema = new mongoose.Schema(
+  {
+    // orgUser_id: { type: String },
```
[the remark this answers is not in the document]
Comment (tlchambers) [reply backend/models/commentModel.js:9]: We agreed to not let user delete a comment for accountability? i.e. if an individual comments saying they want to help but then change their mind... we don't want to give them the option of deleteing?

File: backend/controllers/orgUsersController.js (hunk 4 above)
Comment (tlchambers) [reply backend/controllers/orgUsersController.js]: Hmm...  main doesn't have getAllOrgUsers  - do you want me to readd it back?

File: backend/spec/controllers/orgUsersController.spec.js [hunk 7]
```diff
@@ -1,28 +1,261 @@
...
+        organisationName: "Puppies Trust",
+        email: "puppy@gmail.com",
+        charityNumber: 123456,
+        password: "ThisIsAStrongPassword1!",
+      });
+      expect(response.statusCode).toBe(201);
+    });
   });
+
+  describe("Throws error meessages when the client already has an account", () => {
```
Comment (Sabirah42) [inline backend/spec/controllers/orgUsersController.spec.js:167]: Small typo in 'messages'

File: backend/spec/models/orgUserModel.spec.js [hunk 8]
```diff
@@ -1,59 +1,75 @@
...
+    it("returns error message when any of the fields that are 'required' is left blank", () => {
+      const orgUser = new OrgUser({
+        organisationName: "",
+        email: "",
+        charityNumber: "123456",
+        password: "",
+      });
+      expect(orgUser.error).not.toEqual("");
     });
-    expect(orgUser.charityNumber).toEqual(null);
```
Comment (Sabirah42) [inline backend/spec/models/orgUserModel.spec.js:57]: Are we testing this elsewhere (that the account is created even if charity number is left blank)?

File: backend/models/commentModel.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (Sabirah42) [reply backend/models/commentModel.js:9]: Sorry, I wasn't clear at all!! On line 9, there is commented out code, should this just be deleted to make it tidy?

**Review by Sabirah42 — APPROVED** [review]
Please do take a look at my final comment re. null charity number value, but happy to approve and merge in the interim! :)

File: backend/models/commentModel.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (tlchambers) [reply backend/models/commentModel.js:9]: ah yes!! Working on tidying up code now

File: backend/spec/controllers/orgUsersController.spec.js (hunk 7 above)
Comment (tlchambers) [reply backend/spec/controllers/orgUsersController.spec.js:167]: ahh messages on purpose as the describe is context - it throw didn't error messages. So I will chance throws to throw

File: backend/spec/models/orgUserModel.spec.js (hunk 8 above)
Comment (tlchambers) [reply backend/spec/models/orgUserModel.spec.js:57]: Yes we tested that an orgUser can created an account with charity number and another test where orgUser can create account without charity number

**Pull request merged**